lib/Transform/IR/SplitBasicBlocksWithCallInst.cpp

```diff
@@ -0,0 +1,92 @@
+//===--- SplitBasicBlocksWithCallInst.cpp --- Split Basic Block Tranform ----------*- C++ -*-===//
+//
+//                       Traits Static Analyzer (SAPFOR)
+//
+// Copyright 2018 DVM System Group
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements passe to do split basik block on call instr.
+//
+//===----------------------------------------------------------------------===//
+#include <llvm/ADT/STLExtras.h>
+#include <llvm/Support/Debug.h>
+#include <llvm/Support/raw_ostream.h>
+#include <llvm/Analysis/LoopInfo.h>
+#include <llvm/IR/BasicBlock.h>
+#include <tsar/Transform/IR/SplitBasicBlocksWithCallInst.h>
+
+using namespace llvm;
+
+#undef DEBUG_TYPE
+#define DEBUG_TYPE "Split-BB"
+
+char SplitBasicBlocksWithCallInstPass::ID = 0;
+INITIALIZE_PASS_BEGIN(SplitBasicBlocksWithCallInstPass, "Split-BB",
+  "Split BB in call instraction", false, true)
```

Comment:
Это не проход анализа.
```suggestion
  "Split BB in call instraction", false, false)
```